src/lib.rs

```diff
@@ -35,6 +35,20 @@ use colored::*;
 use log::{Level, LevelFilter, Log, Metadata, Record, SetLoggerError};
 use std::collections::HashMap;
 
+macro_rules! LOGGING_FMT {
+    () => {
+        "{ts}{lvl}{tgt}{thr}{msg}"
+    };
+}
+
+// (prefix, suffix)
+#[cfg(feature = "chrono")]
+const TS_FMT: (&str, &str) = ("", " ");
+const LVL_FMT: (&str, &str) = ("", " ");
+const TGT_FMT: (&str, &str) = ("[", "] ");
+#[cfg(feature = "thread_ids")]
+const TRD_FMT: (&str, &str) = ("{", "} ");
```

Comment:
I'm not sure these need to be constants (it might be better to keep these strings with the `Field::new()` calls), but if they stay up here could we use full names for them instead (e.g. `THREAD_FORMAT`)?

Reply:
If fields always have a trailing space it might be best to make that part of the Display implmentation for Field instead of putting it in these constants, it's quite easy to miss here.

Reply:
1. I wanted the log format configuration at one place, eventually(!) configurable by features (feature = foo_format | bar_format, custom_format). Scatter this back in the code will lead again to a lot more #[cfg(...)] mess. Actually it would be nice when const generics could handle &str, but we are not there yet, thats for future.

2. Having prefix/postfix is the most generic way to enable all kinds of possible log formats, for example including my first attempt with [thread@target]. The trailing space is not necessary there. Addings an extra 'separator' would add more weight while postfix can all ready do that.

Reply:
I hate to keep being negative as this is really good work, but being able to configure the output format is something I wouldn't want to merge into this library. There's much better rust libraries out there for configurable logging, and I see the value of `simple_logger` as something minimal that "just works" without the user needing to worry about configuration. Most of the existing config options are only there so users can avoid pulling in extra dependencies.

Reply:
Maybe I can convince you some day in future (this configurability isn't my primary goal anyway). First and foremost this should stay simple and easily maintainable and "just work" as you say. That is exactly the reason why i use simple_logger. Also i do not really want to fork 'just another logging lib' rather contribute here, make it more useful without frills.